### PR TITLE
Copy list settings instances

### PR DESCRIPTION
**Pull request by yanaise**
@adamro This is my commit for uploading settings. I would appreciate a cr.
Note that currently I only upload settings of the "list" type, because Hierarchical settings raise different issues.

### CODE REVIEW

File: packages/salesforce-adapter/src/filters/custom_settings_filter.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { Element, isObjectType } from '@salto-io/adapter-api'
+import _ from 'lodash'
+import { FilterCreator } from '../filter'
+import { isCustomSettingsObject, apiName } from '../transformers/transformer'
+import { ConfigChangeSuggestion } from '../types'
+import { getAllInstances } from './custom_objects_instances'
+
+const filterCreator: FilterCreator = ({ client }) => ({
+  onFetch: async (elements: Element[]): Promise<ConfigChangeSuggestion[]> => {
+    const customSettingsFetchSettings = elements.filter(obj => (isCustomSettingsObject(obj)
```
Comment (adamro) [inline packages/salesforce-adapter/src/filters/custom_settings_filter.ts:25]: You have this `isCustomSettingsObject(obj) && obj.annotations.customSettingsType === 'List'` more than once, extract to a func?
Also, use `obj.annotations[CUSTOM_SETTINGS_TYPE]`

File: packages/salesforce-adapter/src/custom_object_instances_deploy.ts [hunk 2]
```diff
@@ -116,6 +117,15 @@ const getRecordsBySaltoIds = async (
   return (await toArrayAsync(recordsIterable)).flat()
 }
 
+const getDataManagementConfigIfCustomSettings = (instances: InstanceElement[]):
+  DataManagementConfig | undefined => ((isCustomSettingsObject(instances[0].type)
+    && instances[0].type.annotations[CUSTOM_SETTINGS_TYPE] === 'List') ? ({
```
[the remark this answers is not in the document]
Comment (yanaise) [reply packages/salesforce-adapter/src/custom_object_instances_deploy.ts:122]: if they have the same type, how can their type have different annotations?

[the remark this answers is not in the document]
Comment (adamro) [reply packages/salesforce-adapter/src/custom_object_instances_deploy.ts:122]: Well line 303 only checks for single apiName and doesn't actually check full equality but yea I guess you are right and there shouldn't be a use case for diff annotations here

File: packages/salesforce-adapter/src/custom_object_instances_deploy.ts [hunk 3]
```diff
@@ -289,16 +298,19 @@ export const deployCustomObjectInstancesGroup = async (
...
-    if (dataManagementConfig === undefined) {
-      throw new Error('dataManagement must be defined in the salesforce.nacl config to deploy Custom Object instances')
-    }
     const instances = changeGroup.changes.map(change => getChangeElement(change))
     const instanceTypes = [...new Set(instances.map(inst => apiName(inst.type)))]
     if (instanceTypes.length > 1) {
       throw new Error(`Custom Object Instances change group should have a single type but got: ${instanceTypes}`)
     }
+    const actualDataManagement = (isCustomSettingsObject(instances[0].type)
+    && instances[0].type.annotations[CUSTOM_SETTINGS_TYPE] === 'List') ? getDataManagementConfigForCustomSettings(instances)
```
Comment (adamro) [inline packages/salesforce-adapter/src/custom_object_instances_deploy.ts:307]: I would still extract the `isCustomSettingsObject(type) && type.annotations[CUSTOM_SETTINGS_TYPE] === 'List')` to an exported func from the `custom_settings_filter`

**Review by adamro — APPROVED** [review]

**Pull request merged**